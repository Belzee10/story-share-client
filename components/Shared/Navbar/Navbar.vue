<template>
  <v-container>
    <v-app-bar flat height="100">
      <v-row>
        <v-col cols="4" offset="4" class="d-flex justify-center">
          <v-img
            :src="require('~/assets/images/logo.png')"
            max-width="180"
          ></v-img>
        </v-col>
        <v-col cols="4" class="d-flex justify-end align-center">
          <v-btn color="primary" class="text-capitalize">
            Get Started
          </v-btn>
        </v-col>
      </v-row>
      <template v-slot:extension>
        <v-row>
          <v-col
            :class="getExtensionbarStyles"
            class="d-flex justify-center py-2"
          >
            <nuxt-link
              v-for="item in getLinks"
              :key="item.id"
              :to="item.to"
              class="category-link"
            >
              {{ item.name }}
            </nuxt-link>
          </v-col>
        </v-row>
      </template>
    </v-app-bar>
  </v-container>
</template>

<script>
import { mapActions, mapState } from 'vuex';

export default {
  name: 'Navbar',
  computed: {
    ...mapState('Categories', ['categories']),
    getExtensionbarStyles() {
      return !this.$vuetify.theme.dark ? 'border' : '';
    },
    getLinks() {
      const links = this.categories.map(item => ({
        ...item,
        to: item.name.toLowerCase()
      }));
      return [{ id: 'all', name: 'All', to: '/' }, ...links];
    }
  },
  mounted() {
    this.getCategories();
  },
  methods: {
    ...mapActions('Categories', ['getCategories'])
  }
};
</script>

<style lang="scss" scoped>
.border {
  border-top: 1px solid var(--v-accent-base);
  border-bottom: 1px solid var(--v-accent-base);
}
a {
  text-decoration: none;
  margin-right: 1.5rem;
  text-transform: uppercase;
}
a:hover {
  color: var(--v-anchor-lighten);
}
</style>
