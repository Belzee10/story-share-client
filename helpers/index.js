/**
 * transform the response from express
 * remove values from exclude
 * change the key "_id" by "id"
 * @param {Object} item
 * @param {Array} exclude
 */
export const transformResponse = (item, exclude = ['__v', 'created_at']) => {
  Object.keys(item).forEach(key => {
    if (exclude.includes(key)) delete item[key];
  });
  if (item._id) {
    item.id = item._id;
    delete item._id;
  }
  return item;
};
