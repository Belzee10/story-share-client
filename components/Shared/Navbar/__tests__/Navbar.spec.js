import provokeArray from 'provoke-array';
import { Store } from 'vuex-mock-store';
import Navbar from '../Navbar';
import { createWrapper } from '~/config/tests.helpers';

describe('Navbar.vue', () => {
  const categories = provokeArray(3, i => ({ name: `Name: ${i}` }));
  const store = new Store({
    state: {
      Categories: {
        categories
      }
    }
  });
  test('should render all the "categories"', () => {
    const wrapper = createWrapper(Navbar, { store });
    expect(wrapper.findAll('.category-link')).toHaveLength(
      categories.length + 1
    );
  });

  test('should render correctly', () => {
    const wrapper = createWrapper(Navbar, { store });
    expect(wrapper.element).toMatchSnapshot();
  });
});
