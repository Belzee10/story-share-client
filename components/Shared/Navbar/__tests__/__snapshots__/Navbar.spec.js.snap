// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue should render correctly 1`] = `
<div
  class="container"
>
  <!---->
   
  <header
    class="v-sheet v-sheet--tile theme--light v-toolbar v-toolbar--extended v-toolbar--flat v-app-bar"
    style="height: 148px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 100px;"
    >
      <div
        class="row"
      >
        <div
          class="d-flex justify-center col col-4 offset-4"
        >
          <nuxtlink
            to="/"
          >
            <div
              class="v-responsive v-image"
              style="max-width: 180px;"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </nuxtlink>
        </div>
         
        <div
          class="d-flex justify-end align-center col col-4"
        >
          <button
            class="text-capitalize v-btn v-btn--depressed theme--light v-size--default primary"
            type="button"
          >
            <span
              class="v-btn__content"
            >
              
          Get Started
        
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="v-toolbar__extension"
      style="height: 48px;"
    >
      <div
        class="row"
      >
        <div
          class="d-flex justify-center py-2 col border"
        >
          <nuxt-link
            class="category-link"
            to="/"
          >
            
            All
          
          </nuxt-link>
          <nuxt-link
            class="category-link"
            to="name: 0"
          >
            
            Name: 0
          
          </nuxt-link>
          <nuxt-link
            class="category-link"
            to="name: 1"
          >
            
            Name: 1
          
          </nuxt-link>
          <nuxt-link
            class="category-link"
            to="name: 2"
          >
            
            Name: 2
          
          </nuxt-link>
        </div>
      </div>
    </div>
  </header>
</div>
`;
